### PR TITLE
Workaround for yast2 crash while tests running on openQA

### PR DESCRIPTION
**Pull request by Zaoliang**
- add record_soft_failure "bsc#1049433"
- please see also ticket for further information:
  https://progress.opensuse.org/issues/23616

### CODE REVIEW

**Comment by Zaoliang** [comment]
This issue happened on openQA production server only. 
My test reference can be found at: http://e13.suse.de/tests/4002

File: tests/console/yast2_xinetd.pm [hunk 1]
```diff
@@ -27,7 +27,11 @@ sub run {
     script_run("yast2 xinetd; echo yast2-xinetd-status-\$? > /dev/$serialdev", 0);
 
     # check xinetd network configuration got started
-    assert_screen 'yast2_xinetd_startup', 90;
+    assert_screen([qw(yast2_xinetd_startup yast2_xinetd_core-dumped)]), 90;
+    if (match_has_tag('yast2_xinetd_core-dumped')) {
+        # softfail when yast2 crashed and throws core dumped message
+        record_soft_failure "bsc#1049433";
+    }
```
Comment (okurz) [inline tests/console/yast2_xinetd.pm]: You are not doing anything special when you match that tag. Why not just create a workaround needle with the tag "yast2_xinetd_startup" referencing the bug id?

File: tests/console/yast2_xinetd.pm [hunk 2]
```diff
@@ -27,7 +27,11 @@ sub run {
     script_run("yast2 xinetd; echo yast2-xinetd-status-\$? > /dev/$serialdev", 0);
 
     # check xinetd network configuration got started
-    assert_screen 'yast2_xinetd_startup', 90;
+    assert_screen([qw(yast2_xinetd_startup yast2_xinetd_core-dumped)]), 90;
+    if (match_has_tag('yast2_xinetd_core-dumped')) {
+        # softfail when yast2 crashed and throws core dumped message
+        record_soft_failure "bsc#1049433";
```
Comment (okurz) [inline tests/console/yast2_xinetd.pm]: just add `return 1;` to make the whole test step soft fail and *not* fail. But if you do this then no logs are collected or uploaded. Better to call the post_fail_hook explicitly and then return, e.g. (untested):

Add in line 20:

```perl
my $self = @_;
```

```perl
record_soft_failure "bsc#1049433";
$self->SUPER::post_fail_hook();
return 1;
```

you can test that with a test needle that matches even normal passing windows but the tag "yast2_xinetd_core-dumped".

Comment (Soulofdestiny) [reply tests/console/yast2_xinetd.pm]: @Zaoliang could you please update? ;)

**Comment by Zaoliang** [comment]
@Soulofdestiny okay, since core dumps could be collected and I will run this on my local server and update this if it works, thanks.

**Pull request closed without merging**